rollup/types/encoding/codecv1/codecv1.go

```diff
@@ -291,63 +283,79 @@ func computeBatchDataHash(chunks []*encoding.Chunk, totalL1MessagePoppedBefore u
 }
 
 // constructBlobPayload constructs the 4844 blob payload.
-func constructBlobPayload(chunks []*encoding.Chunk) (*kzg4844.Blob, common.Hash, *kzg4844.Point, error) {
+func constructBlobPayload(chunks []*encoding.Chunk, useCompression bool) (*kzg4844.Blob, common.Hash, *kzg4844.Point, error) {
+	maxNumChunks := getMaxNumChunks(useCompression)
+
+	// this encoding can only support a fixed number of chunks per batch
+	if len(chunks) > maxNumChunks {
+		return nil, common.Hash{}, nil, fmt.Errorf("too many chunks in batch")
+	}
+
 	// metadata consists of num_chunks (2 bytes) and chunki_size (4 bytes per chunk)
-	metadataLength := 2 + MaxNumChunks*4
+	metadataLength := 2 + maxNumChunks*4
 
-	// the raw (un-padded) blob payload
-	blobBytes := make([]byte, metadataLength)
+	// the raw (un-padded) batch payload
+	batchBytes := make([]byte, metadataLength)
 
 	// challenge digest preimage
 	// 1 hash for metadata, 1 hash for each chunk, 1 hash for blob versioned hash
-	challengePreimage := make([]byte, (1+MaxNumChunks+1)*32)
+	challengePreimage := make([]byte, (1+maxNumChunks+1)*32)
 
 	// the chunk data hash used for calculating the challenge preimage
 	var chunkDataHash common.Hash
 
-	// blob metadata: num_chunks
```

Comment:
Let's keep this code the same as the monorepo version, we can leave the comments as `blob *`

Reply:
I also changed the comment in this PR: https://github.com/scroll-tech/scroll/pull/1298
currently this PR just syncs the implementation in the above PR.